src/almanack/metrics/data.py

```diff
@@ -1182,3 +1195,71 @@ def find_doi_citation_data(repo: pygit2.Repository) -> Dict[str, Any]:
                 LOGGER.warning(f"Error during OpenAlex exact DOI lookup: {e}")
 
     return result
+
+
+def compute_sustainability_score(
+    table: List[Dict[str, Union[int, float, bool]]]
+) -> float:
+    """
+    Computes a sustainability score by normalizing
+    numeric metrics and incorporating boolean metrics.
+    Adjusts for metrics that should be inversely
+    proportional to the sustainability score.
+
+    Args:
+        data (List[Dict[str, Union[int, float, bool]]]):
+            A list of dictionaries containing metrics.
+            Each dictionary must have a "result" key
+            with a value that is an int, float, or bool.
+              Optionally, a "direction" key may be
+            included for numeric values to specify the
+            relationship to sustainability:
+            - 1 (positive correlation)
+            - 0 (no correlation)
+            - -1 (negative correlation)
+
+    Returns:
+        float:
+            The computed sustainability score, normalized between 0 and 1.
+    """
+    results = []
+    bool_results = []
+
+    # Gather numeric and boolean values with a direction
+    for item in table:
+        if isinstance(item["result"], bool) and item["direction"] != 0:
+            # for direction == 1 we treat True as positive correction
+            # and False as a negative correlation.
+            if item["direction"] == 1:
+                bool_results.append(1 if item["result"] else 0)
+            # for direction == -1 we treat True as negative correction
+            # and False as a positive correlation.
+            elif item["direction"] == -1:
+                bool_results.append(0 if item["result"] else 1)
```

Comment:
Perhaps an example would add a bit more clarity?

Also, what happens if there is no `direction` key in item? (I don't see an exception raised or logic on what to do in this scenario). Alternatively, you could also include some data validation logic prior to this for loop

Reply:
Thanks - good call. I'll add an example here. The way the schema is composed within `metrics.yml` along with the jsonschema checks in the tests we now require that all metrics have a direction (1, -1, or 0). I think this is a necessary step to help classify the metrics while retaining their distinct information.